packages/cdktf/lib/terraform-functions.ts

```diff
@@ -0,0 +1,1221 @@
+        );
+      }
+      args.forEach((arg, i) => argValidators[i](arg));
+      return call(name, args);
+    } else {
+      return call(name, argValidators(args));
+    }
+  };
+}
+
+export class Fn {
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/anytrue.html anytrue} returns true if all elements in a given collection are true or "true"
+   * @param {Array} values
+   */
+  public static alltrue(values: any[]) {
+    return terraformFunction("alltrue", [listOf(anyValue)])(values);
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/anytrue.html anytrue} returns true if any element in a given collection is true or "true"
+   * @param {Array} value
+   */
+  public static anytrue(value: any[]) {
+    return terraformFunction("anytrue", [listOf(anyValue)])(value);
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/chunklist.html chunklist} splits a single list into fixed-size chunks, returning a list of lists.
+   * @param {Array} value
+   * @param {number} chunkSize
+   */
+  public static chunklist(value: any[], chunkSize: number | IResolvable) {
+    return asList(
+      terraformFunction("chunklist", [listOf(anyValue), numericValue])(
+        value,
+        chunkSize
+      )
+    );
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/coalesce.html coalesce} takes any number of arguments and returns the first one that isn't null or an empty string.
+   * @param {Array} value - Arguments are passed in an array
+   */
+  public static coalesce(value: any[]) {
+    return terraformFunction("coalesce", listOf(anyValue))(...value);
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/coalescelist.html coalescelist} takes any number of list arguments and returns the first one that isn't empty.
+   * @param Array} value - Arguments are passed in an array
+   */
+  public static coalescelist(value: any[]) {
+    return terraformFunction("coalescelist", listOf(anyValue))(...value);
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/compact.html compact} takes a list of strings and returns a new list with any empty string elements removed.
+   * @param {Array} value
+   */
+  public static compact(value: any[]) {
+    return asList(terraformFunction("compact", [listOf(anyValue)])(value));
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/concat.html concat} takes two or more lists and combines them into a single list.
+   * @param {Array} value
+   */
+  public static concat(value: any[]) {
+    return asList(terraformFunction("concat", listOf(anyValue))(...value));
+  }
+
+  /**
+   * {@link https://www.terraform.io/docs/language/functions/contains.html contains} determines whether a given list or set contains a given single value as one of its elements.
+   * @param {Array} list
+   * @param {any} value
+   */
+  public static contains(list: any[] | string, value: any) {
```

Comment:
I didn't see `contains` listed under string functions; does it accept a string?
We don't have the set type currently, but would be nice to make a note of it for the future.